thrift.js

```diff
@@ -123,17 +162,58 @@ Thrift.prototype._definitionProcessors = {
     Union: 'compileUnion'
 };
 
+Thrift.prototype.compileHeaders = function compileHeaders(defs) {
+    var self = this;
+    for (var index = 0; index < defs.length; index++) {
+        var def = defs[index];
+        var headerProcessor = self._headerProcessors[def.type];
+        // istanbul ignore else
+        if (headerProcessor) {
+            self[headerProcessor](def);
+        }
+    }
+};
+
 Thrift.prototype.compileDefinitions = function compileDefinitions(defs) {
     var self = this;
     for (var index = 0; index < defs.length; index++) {
         var def = defs[index];
+        var definitionProcessor = self._definitionProcessors[def.type];
         // istanbul ignore else
-        if (self._definitionProcessors[def.type]) {
-            self[self._definitionProcessors[def.type]](def);
+        if (definitionProcessor) {
+            self[definitionProcessor](def);
         }
     }
 };
 
+Thrift.prototype.compileInclude = function compileInclude(def) {
+    var self = this;
+
+    if (def.id.indexOf('./') === 0 || def.id.indexOf('../') === 0) {
```

Comment:
Doesn't seem to make that big a difference per jsPerf, but resolved anyways:

https://jsperf.com/indexof-vs-lastindexof-with-0-start-index